src/openms/source/QC/DBSuitability.cpp

```diff
@@ -290,9 +229,320 @@ namespace OpenMS
     return true;
   }
 
-  bool DBSuitability::passesFDR_(const PeptideHit& hit, double FDR)
+  bool DBSuitability::passesFDR_(const PeptideHit& hit, double FDR) const
   {
     if (hit.getScore() > FDR) return false;
     return true;
   }
-}
+
+  pair<String, Param> DBSuitability::extractSearchAdapterInfoFromMetaValues_(const ProteinIdentification::SearchParameters& search_params) const
+  {
+    Param p;
+    // list of all allowed adapters
+    vector<String> working_adapters{ "CometAdapter", "CruxAdapter", "MSGFPlusAdapter", "MSFraggerAdapter", "MyriMatchAdapter", "OMSSAAdapter", "XTandemAdapter" };
+
+    String adapter;
+    vector<String> keys;
+    search_params.getKeys(keys);
+    for (const String& key : keys)
+    {
+      if (adapter.empty()) // trying to find which adapter was used
+      {
+        for (const String& a : working_adapters)
+        {
+          if (key.find(a) == 0)
+          {
+            // used adapter found
+            adapter = a;
+            break;
+          }
+        }
+      }
+      else // found adapter name
+      {
+        if (key.find(adapter) != 0) continue; // does adapter appear in meta value key?
+        p.setValue(key, search_params.getMetaValue(key));
+      }
+    }
+
+    if (p.empty()) // non of the allowed adapter names where found in the meta values
+    {
+      String message;
+      message = "No parameters found for any of the allowed adapters in the given meta values. Allowed are:\n";
+      for (const String& a : working_adapters)
+      {
+        message += a + " ";
+      }
+      message = "\n";
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, message);
+    }
+
+    OPENMS_LOG_DEBUG << "Parameters for the following adapter were found: " << adapter << endl;
+
+    return make_pair(adapter, p);
+  }
+
+  void DBSuitability::writeIniFile_(const Param& parameters, const String& filename) const
+  {
+    if (!File::writable(filename)) // check if file exists and is writable
+    {
+      throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Unable to create file: " + filename);
+    }
+    ParamXMLFile param_file;
+    param_file.store(filename, parameters);
+  }
+
+  vector<PeptideIdentification> DBSuitability::runIdentificationSearch_(const MSExperiment& exp, const vector<FASTAFile::FASTAEntry>& fasta_data, const String& adapter_name, Param& parameters) const
+  {
+    if (adapter_name.empty())
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No adapter name given. Aborting!");
+    }
+
+    // temporary folder for search in- und output files
+    File::TempDir tmp_dir;
+    String mzml_path = tmp_dir.getPath() + "spectra.mzML";
+    String db_path = tmp_dir.getPath() + "database.FASTA";
+    String out_path = tmp_dir.getPath() + "out.idXML";
+
+    // override the in- and output files in the parameters
+    parameters.setValue(adapter_name + ":1:in", mzml_path);
```

Comment:
sanity check: make sure these parameters exist. Otherwise you create new ones which will be ignored (or crash the tool)